oem/ibm/libpldmresponder/file_io_type_dump.cpp

```diff
@@ -426,20 +462,63 @@ int DumpHandler::readIntoMemory(uint32_t offset, uint32_t& length,
                                 uint64_t address,
                                 oem_platform::Handler* /*oemPlatformHandler*/)
 {
-    if (dumpType != PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    auto path = findDumpObjPath(fileHandle);
+    static constexpr auto dumpFilepathInterface =
+        "xyz.openbmc_project.Common.FilePath";
+    if (dumpType == PLDM_FILE_TYPE_DUMP)
     {
         return PLDM_ERROR_UNSUPPORTED_PLDM_CMD;
     }
+    else if (dumpType != PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    {
+        try
+        {
+            auto filePath =
+                pldm::utils::DBusHandler().getDbusProperty<std::string>(
+                    path.c_str(), "Path", dumpFilepathInterface);
+            auto rc = transferFileData(fs::path(filePath), false, offset,
+                                       length, address);
+            return rc;
+        }
+        catch (const std::exception& e)
+        {
+            std::cerr << "Failed to fetch the filepath of the dump entry"
+                      << std::hex << fileHandle << ", error = " << e.what()
+                      << "\n";
+            return PLDM_ERROR;
+        }
+    }
     return transferFileData(resDumpDirPath, true, offset, length, address);
 }
 
 int DumpHandler::read(uint32_t offset, uint32_t& length, Response& response,
                       oem_platform::Handler* /*oemPlatformHandler*/)
 {
-    if (dumpType != PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    auto path = findDumpObjPath(fileHandle);
+    static constexpr auto dumpFilepathInterface =
+        "xyz.openbmc_project.Common.FilePath";
+    if (dumpType == PLDM_FILE_TYPE_DUMP)
```

Comment:
This has changed the old logic. Add an || with PLDM_FILE_TYPE_RESOURCE_DUMP too.